### PR TITLE
Check  for programs with CMAKE_INSTALL_PREFIX

### PR DESCRIPTION
**Pull request by daniellowell**
Fixes issue: [SWDEV-269156](http://ontrack-internal.amd.com/browse/SWDEV-269156)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -149,15 +149,19 @@ if( MIOPEN_BACKEND STREQUAL "OpenCL")
     find_package( OpenCL REQUIRED )
     find_program(MIOPEN_HIP_COMPILER hcc
             PATH_SUFFIXES bin
-            PATHS /opt/rocm
+            PATHS 
+	        /opt/rocm
+		${CMAKE_INSTALL_PREFIX}
```
Comment (pfultz2) [inline CMakeLists.txt:154]: CMake already searches for in the `CMAKE_INSTALL_PREFIX`, so this isn't needed here.

File: CMakeLists.txt [hunk 2]
```diff
@@ -149,15 +149,19 @@ if( MIOPEN_BACKEND STREQUAL "OpenCL")
...
     set(MIOPEN_USE_MIOPENGEMM ON CACHE BOOL "")
     if(MIOPEN_HIP_COMPILER)
         message("hip compiler: ${MIOPEN_HIP_COMPILER}")
     else()
         find_program(MIOPEN_HIP_COMPILER clang++
             PATH_SUFFIXES bin
-	        PATHS /opt/rocm/llvm
+	        PATHS 
+		    /opt/rocm/llvm
+		    ${CMAKE_INSTALL_PREFIX}/llvm
```
Comment (pfultz2) [inline CMakeLists.txt:164]: So the `/opt/rocm/llvm` can be removed as its already added to the `CMAKE_PREFIX_PATH` on line 110. We should probably add `${CMAKE_INSTALL_PREFIX}/llvm` to the CMAKE_PREFIX_PATH as well and remove adding these extra paths everywhere.

File: CMakeLists.txt (hunk 1 above)
Comment (daniellowell) [reply CMakeLists.txt:154]: But it fails to.

Comment (daniellowell) [reply CMakeLists.txt:154]: Scratch that, just for clang-offload-bundler.

Comment (atamazov) [reply CMakeLists.txt:154]: But SWDEV-269156 is about extractkernels isn't it? And this is 4.0.x, so we are not using extractkernels, so why we are looking for it?

Comment (daniellowell) [reply CMakeLists.txt:154]: This is about the symlinks not existing in build dockers. Which means that searching for clang-offload-bundler fails, then the assumption is that we are now in a legacy environment, so MIOpen looks for extractkernel, which also then fails.

Comment (atamazov) [reply CMakeLists.txt]: I see. Usually I ignore that and simply `ln -s /opt/rocm-... /opt/rocm` but this is inconvenient when one works with multiple installations.

**Pull request merged**